app/models/cart.rb

```diff
@@ -0,0 +1,18 @@
+class Cart < ActiveRecord::Base
+  has_many :line_items, dependent: :destroy
+
+  def add_product(product_id)
+    current_item = line_items.find_by_product_id(product_id)
```

Comment:
try to line item in one query
current_item = LineItem.find_by_product_id_and_cart_id(product_id, self.id)